src/vanilla/utils/atomWithStorage.ts

```diff
@@ -5,8 +5,21 @@ import { RESET } from './constants.ts'
 const isPromiseLike = (x: unknown): x is PromiseLike<unknown> =>
   typeof (x as any)?.then === 'function'
 
+type Subscribe<Value> = (
+  key: string,
+  callback: (value: Value) => void,
+  initialValue: Value,
+) => Unsubscribe
+
 type Unsubscribe = () => void
 
+type SubscribeHandler<Value> = (
+  subscribe: Subscribe<Value>,
+  key: string,
+  callback: (value: Value) => void,
+  initialValue: Value,
+) => Unsubscribe
+
```

Comment:
Maybe, we don't need this type alias. (But, not 100% sure.)

Reply:
would something like this be more acceptable?:

```ts
const handleSubscribe = (
    subscriber: Subscribe<Value>,
    ...params: Parameters<Subscribe<Value>>
  ) => {
    const [key, callback, initialValue] = params

    function callbackWithParser(v: Value) {
      let newValue: Value
      try {
        newValue = JSON.parse((v as string) || '')
      } catch {
        newValue = initialValue
      }

      callback(newValue as Value)
    }

    return subscriber(key, callbackWithParser, initialValue)
  }
```

Reply:
Yes, I meant to inline it as it's used just once.